tests/data/test_delta_dataset_source.py

```diff
@@ -113,3 +115,108 @@ def test_delta_dataset_source_too_many_inputs(spark_session, tmp_path):
 
     with pytest.raises(MlflowException, match='Must specify exactly one of "path" or "table_name"'):
         DeltaDatasetSource(path=tmp_path, delta_table_name="temp_delta_too_many_inputs")
+
+
+def test_uc_table_id_retrieval_works(spark_session, tmp_path):
+    def mock_resolve_table_name(table_name):
+        if table_name == "temp_delta_versioned_with_id":
+            return "default.temp_delta_versioned_with_id"
+        return table_name
+
+    def mock_lookup_table_id(table_name):
+        if table_name == "default.temp_delta_versioned_with_id":
+            return "uc_table_id_1"
+        return None
+
+    def mock_is_databricks_uc_table():
+        return True
+
+    with mock.patch(
+        "mlflow.data.delta_dataset_source.DeltaDatasetSource._resolve_table_name",
+        side_effect=mock_resolve_table_name,
+    ):
+        with mock.patch(
+            "mlflow.data.delta_dataset_source.DeltaDatasetSource._lookup_table_id",
+            side_effect=mock_lookup_table_id,
+        ):
+            with mock.patch(
+                "mlflow.data.delta_dataset_source.DeltaDatasetSource._is_databricks_uc_table",
+                side_effect=mock_is_databricks_uc_table,
+            ):
+                df = pd.DataFrame([[1, 2, 3], [1, 2, 3]], columns=["a", "b", "c"])
+                df_spark = spark_session.createDataFrame(df)
+                df_spark.write.format("delta").mode("overwrite").saveAsTable(
+                    "default.temp_delta_versioned_with_id", path=tmp_path
+                )
+
+                df2 = pd.DataFrame([[1, 2, 3]], columns=["a", "b", "c"])
+                df2_spark = spark_session.createDataFrame(df2)
+                df2_spark.write.format("delta").mode("overwrite").saveAsTable(
+                    "default.temp_delta_versioned_with_id", path=tmp_path
+                )
+
+                delta_datasource = DeltaDatasetSource(
+                    delta_table_name="temp_delta_versioned_with_id", delta_table_version=1
+                )
+                loaded_df_spark = delta_datasource.load()
+                assert loaded_df_spark.count() == df2_spark.count()
+                assert delta_datasource.to_json() == json.dumps(
+                    {
+                        "delta_table_name": "default.temp_delta_versioned_with_id",
+                        "delta_table_version": 1,
+                        "is_databricks_uc_table": True,
+                        "delta_table_id": "uc_table_id_1",
+                    }
+                )
+
+
+def test_uc_table_id_retrieval_throws(spark_session, tmp_path):
+    def mock_resolve_table_name(table_name):
+        if table_name == "temp_delta_versioned_with_id_throws":
+            return "default.temp_delta_versioned_with_id_throws"
+        return table_name
+
+    def mock_lookup_table_id(table_name):
+        if table_name == "default.temp_delta_versioned_with_id_throws":
```

Comment:
It doesn't seem like we're testing that ID retrieval throws here

Reply:
Yeah I'm realizing that with the broken test result.  This was the last thing I had done last night so wasn't clearly thinking this through.  Will fix.